### PR TITLE
Kait's HW 12/8

### PR DESCRIPTION
**Pull request by kmikitin**

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+// reducer is taking 2 numbers in the array, adding them to each other and then adding to current value, and so on; this is similar to how a for loop works. NEED THIS FUNCTION TO USE .reduce().
+
+function sumDigits(number) {
+	// turn the number into an array of WHOLE integers
+	// How: make a string, split the string, use . map to make numbers (from library)
+	let numArray = number.toString(10).split("").map(Number);
+	// 10 = radix number (so 0-9)
+	return numArray.reduce(reducer);
+}
+
```
Comment (reubenayres) [inline js/app.js:32]: You dove deep into some Array functions! Nice. This bit is intriguing: `… .map(Number);`

File: js/app.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	// divide num by all numbers up to num -1
+	for (i = 2; i < num - 1; i++) {
+		// if num is divisible by ANY but 1 and itself return false
+		if (num % i == 0) {
+		return false
+		} 
+	} 
+	return true 
+}
+
```
Comment (reubenayres) [inline js/app.js:70]: this says that 1 and 0 are prime, which they are not (although for admittedly very abstract theoretical math reasons)
also negative numbers...
consider edge case tests

File: js/app.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+// Step Two
+function printPrimes(limit){
+    // loop through all numbers up to limit
+    // start at 2, since 1 is not prime
+    for(var i = 2; i <= limit; i++ ) { 
+    // call checkPrime to see if each iteration is prime
+      if(checkPrime(i)===true) {
+          // Print prime numbers
```
Comment (reubenayres) [inline js/app.js]: yep or you could just say if(checkPrime(i))

File: js/app.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	return true 
+}
+
+// Step Two
+function printPrimes(limit){
+    // loop through all numbers up to limit
+    // start at 2, since 1 is not prime
+    for(var i = 2; i <= limit; i++ ) { 
+    // call checkPrime to see if each iteration is prime
+      if(checkPrime(i)===true) {
```
Comment (reubenayres) [inline js/app.js:77]: yep or you could just say if(checkPrime(i))

File: js/app.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          // Print prime numbers
+          console.log(i)
+        }
+    }
+}  
+
+// Insert Dash
+function insertDash(num) {
+	// turn a number into a string array 
+	let numberArray = num.toString().split("").map(Number);
```
Comment (reubenayres) [inline js/app.js:87]: there's that.map(Number) thing again

File: js/app.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	// log to check;
+	// create an array of odd numbers for scientific comparison
+	let oddNumbers = [1, 3, 5, 7, 9];
+	// log to check
+	// create a string to hold numbers once checked
+	let dashString = " ";
```
Comment (reubenayres) [inline js/app.js:93]: start with empty string "" not space " "
